src/plan.rs

```diff
@@ -355,67 +404,153 @@ impl<T: FftNum> FftPlannerScalar<T> {
     }
 
     fn design_fft_with_factors(&mut self, len: usize, factors: PrimeFactors) -> Arc<Recipe> {
-        if let Some(fft_instance) = self.design_butterfly_algorithm(len) {
-            fft_instance
-        } else if factors.is_prime() {
-            self.design_prime(len)
-        } else if len.trailing_zeros() <= MAX_RADIX4_BITS && len.trailing_zeros() >= MIN_RADIX4_BITS
-        {
-            if len.is_power_of_two() {
-                Arc::new(Recipe::Radix4(len))
-            } else {
-                let non_power_of_two = factors
-                    .remove_factors(PrimeFactor {
-                        value: 2,
-                        count: len.trailing_zeros(),
-                    })
-                    .unwrap();
-                let power_of_two = PrimeFactors::compute(1 << len.trailing_zeros());
-                self.design_mixed_radix(power_of_two, non_power_of_two)
-            }
+        let recipes = self.design_all_ffts_with_factors(len, factors);
+        //for recipe in recipes.iter() {
+        //    println!("For {}, cost {} -> {:?}", len, recipe.cost(1), recipe);
+        //}
+
+        let fastest = recipes
+            .iter()
+            .min_by(|x, y| {
+                match x.cost(1).partial_cmp(&y.cost(1)) {
```

Comment:
Would `x.cost(1).partial_cmp(&y.cost(1)).unwrap_or(Ordering::Equal)` work here?

Reply:
Yes, don't know why I didn't do it like that from the start.